### ISSUE TITLE
Touch Screen Inertial Scrolling in ImGui

### ISSUE THREAD

**Issue by skylersaleh** [floooh/sokol#684]
Hi, 

I modified sokol_imgui.h to add touchscreen inertial scrolling so that ImGUI feels a bit better on mobile platforms. You can see the changes required at the below commit: 

https://github.com/skylersaleh/SkyEmu/commit/6473a2987b63c8790b13dbcb91852600a83dad23

And you can test it out on my web build here on a mobile device to see how the changes perform (click the menu icon on the top right to open a panel that has scrolling). 

https://web.skyemu.app/

Is this something that you would be interested in trying to mainline into the main Sokol library? 

Thanks,

-Sky

**Comment by floooh** [comment]
Hmm ok, makes sense that there's no inertia for touch events... on macOS when scrolling withe the touchpad I **do** see inertia, but that's because the scroll input is coming in via 'mousewheel' events, and those seem to have a builtin inertia. But that's not the case for the low-level touch events that sokol_app.h provides.

The whole "touch input story" for Dear ImGui isn't all that great yet at the moment, I'm not completely sure if that's mainly because of Dear ImGui or because of sokol_imgui.h (maybe a mix of both), for instance there are some places in the Dear ImGui input code which rely on 'mouse hover' (e.g. that there's a reported mouse position before a mouse click happens).

I really need to focus for a bit on mobile input, but for now I don't want to put this fix into sokol_imgui.h without first understanding better if there are less "intrusive" options...

**Comment by skylersaleh** [comment]
Yeah, the touch input support in DearImGui could use some improvement. I suppose the right way would be for ImGui to take raw touch input points and handle everything itself. I looked into doing that, but it looked pretty involved (as it would involve change to DearImGui, cimgui, and each of the ImGui backends). 

The approach I took here(mapping the touch drags to scrolls when interacting with a UI element that is not drag-able) was a lot easier, but a bit messy.  That was the big reason I thought I would open an issue and ask if it was something you were interested in, rather than filling a pull request. 

Feel free to take any of the code/ideas I used here when investigating, and if you ever want me to submit a pull request updated to the latest version of sokol_app, let me know.
